test/telemetry/src/main/java/org/opensearch/test/telemetry/tracing/StrictCheckSpanProcessor.java

```diff
@@ -27,58 +28,45 @@ public StrictCheckSpanProcessor() {
 
     }
 
+    private static Map<String, MockSpanData> finishedSpanItems = new ConcurrentHashMap<>();
+
     @Override
     public void onStart(Span span) {
         spanMap.put(span.getSpanId(), Thread.currentThread().getStackTrace());
+        finishedSpanItems.put(span.getSpanId(), toMockSpanData(span));
```

Comment:
May be a better name? since the span is just started and not finished yet.

Reply:
Removed above spanMap and renamed finishedSpanItems to spanMap